tests/components/media_player/test_sonos.py

```diff
@@ -253,7 +253,7 @@ def test_sonos_set_sleep_timer(self, set_sleep_timerMock, *args):
         sonos.setup_platform(self.hass, {}, add_devices_factory(self.hass), {
             'host': '192.0.2.1'
         })
-        device = self.hass.data[sonos.DATA_SONOS].devices[-1]
+        device = list(self.hass.data[sonos.DATA_SONOS].devices)[-1]
```

Comment:
Why are you copying the list here?

Reply:
I'm building the list to get the `[-1]` element but maybe there's also a better way to do that? :-)

Reply:
But isn't devices a list?

Reply:
Ah, now I understand 👍.

In these tests `add_devices` is stubbed so `devices` is now the generator that you suggested I use (by removing square brackets).

Reply:
Aaaaaaah, I see. Didn't realize that.

Fun fact: passing a generator was causing crazy bugs when we just switched to asyncio because we would iterate over the generator inside asyncio, causing I/O to be done inside the event loop. We realized this and now EntityPlatform converts it to a list before passing it to asyncio land.